src/constructs/cloudwatch/ec2-alarms.ts

```diff
@@ -46,6 +51,34 @@ export class Gu5xxPercentageAlarm extends GuAlarm {
       alarmDescription: props.alarmDescription ?? defaultDescription,
       evaluationPeriods: props.numberOfMinutesAboveThresholdBeforeAlarm ?? 1,
     };
-    super(scope, id, alarmProps);
+    super(scope, AppIdentity.suffixText(props, "High5xxPercentageAlarm"), alarmProps);
+  }
+}
+
+/**
+ * Creates an alarm which is triggered whenever there have been several healthcheck failures within a single hour.
+ */
+export class GuUnhealthyHostsAlarm extends GuAlarm {
+  constructor(scope: GuStack, props: GuTargetGroupAlarmProps) {
+    const alarmName = `Unhealthy instances for ${props.app} in ${scope.stage}`;
+    const alarmDescription = `${props.app}'s instances have failed healthchecks several times over the last hour.
+      This typically results in the AutoScaling Group cycling instances and can lead to problems with deployment,
+      scaling or handling traffic spikes.
+
+      Check ${props.app}'s application logs or ssh onto an unhealthy instance in order to debug these problems.`;
+    const alarmProps = {
+      ...props,
+      alarmName: alarmName,
+      alarmDescription: alarmDescription,
+      metric: props.targetGroup.metricUnhealthyHostCount(),
```

Comment:
ASG's have a [`GroupTerminatingInstances` metric](https://docs.aws.amazon.com/autoscaling/ec2/userguide/as-instance-monitoring.html#available-cloudwatch-metrics), which I originally planned on using. Unfortunately it doesn't seem possible to ascertain _why_ an instance is terminating using this metric. Consequently, it's not that helpful for this alarm because expected terminations (e.g. after a Riff-Raff deployment, or when an app scales down after a CPU-based scaling event) look exactly the same as instances being terminated due to failing health checks.